packages/contract/src/contract-factory.ts

```diff
@@ -154,18 +198,141 @@ export default class ContractFactory {
   async deployContract<TContract extends Contract = Contract>(
     deployContractOptions: DeployContractOptions = {}
   ): Promise<DeployContractResult<TContract>> {
-    if (this.bytecode.length > MAX_CONTRACT_SIZE) {
+    const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
+
+    if (this.bytecode.length > maxContractSize) {
       throw new FuelError(
         ErrorCode.CONTRACT_SIZE_EXCEEDS_LIMIT,
-        'Contract bytecode is too large. Max contract size is 100KB'
+        'Contract bytecode is too large. Max contract size is 100KB' // change error
       );
     }
 
     const { contractId, transactionRequest } = await this.prepareDeploy(deployContractOptions);
 
+    const transactionResponse = await account.sendTransaction(transactionRequest);
+
+    const waitForResult = async () => {
+      const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
+      const contract = new Contract(contractId, this.interface, account) as TContract;
+
+      return { contract, transactionResult };
+    };
+
+    return { waitForResult, contractId, transactionId: transactionResponse.id };
+  }
+
+  /**
+   * Deploy a contract with the specified options.
+   *
+   * @param deployContractOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContractLoader<TContract extends Contract = Contract>(
+    deployContractOptions: DeployContractOptions = {}
+  ): Promise<DeployContractResult<TContract>> {
     const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
 
-    const transactionResponse = await account.sendTransaction(transactionRequest);
+    // Ensure the max contract size is byte aligned before chunking
+    if (maxContractSize % WORD_SIZE !== 0) {
```

Comment:
Why is this check our concern? Shouldn't it be the responsibility of the network/node runners to have this be word size aligned?

Reply:
Agreed. However, muck alike other bytcode position issues, if there is an alignment issue it's likely to fail for a different reason (e.g. blob ID not found) which isn't helpful. I'd be okay with removing this though now that things are in the right place.